cogs/handles.py

```diff
@@ -77,72 +78,56 @@ async def link(self, ctx, username: str):
+        )
 
-        await ctx.send(
-            '%s, submit a compiler error to <https://dmoj.ca/problem/%s> '
-            'within 60 seconds' % (ctx.author.mention, problem.code))
-        await asyncio.sleep(60)
-
-        submissions = await query.get_latest_submissions(username, 10)
-
-        # if the user links twice, it might break the db
-        # Should add decorator to prevent this
-        if query.get_handle(ctx.author.id, ctx.guild.id):
-            return
-
-        for submission in submissions:
-            if (submission.result == 'CE' and
-                    submission.problem[0].code == problem.code):
-
-                handle = Handle_DB()
-                handle.id = ctx.author.id
-                handle.handle = username
-                handle.user_id = user.id
-                handle.guild_id = ctx.guild.id
-                session.add(handle)
-                session.commit()
-
-                rank_to_role = {role.name: role for role in ctx.guild.roles if role.name in RANKS}
-                rank = self.rating_to_rank(user.rating)
-                if rank in rank_to_role:
-                    await self._update_rank(ctx.author, rank_to_role[rank], 'Dmoj account linked')
-                else:
-                    await ctx.send("You are missing the " + rank.name + " role")
-
-                return await ctx.send(
-                    "%s, you now have linked your account to %s." %
-                    (ctx.author.name, username)
-                )
-        else:
-            return await ctx.send('I don\'t see anything :monkey: '
-                                  '(Failed to link accounts)')
-
-    @commands.command(name='set', usage='discord_account dmoj_handle')
+    @commands.command(name='set', usage='discord_account [dmoj_handle, +remove]')
     @commands.has_role('Admin')
-    async def _set(self, ctx, member: discord.Member, username: str):
+    async def _set(self, ctx, member, username: str):
+        
         """Manually link two accounts together"""
         query = Query()
-        user = await query.get_user(username)
+        member=await query.parseUser(ctx,member)
 
-        if user is None:
-            await ctx.send(f'{username} does not exist on dmoj')
-            return
+        if username!="+remove":
+            user = await query.get_user(username)
 
-        username = user.username
+            if user is None:
+                await ctx.send(f'{username} does not exist on dmoj')
+                return
 
-        if query.get_handle(member.id, ctx.guild.id):
-            await ctx.send(
-                '%s, this handle is already linked with %s.' %
-                (ctx.author.mention, query.get_handle(member.id, ctx.guild.id)))
+            username = user.username
+
+        handle=query.get_handle(member.id, ctx.guild.id)
+        if handle==username:
+            return await ctx.send(f'{member.display_name} is already linked with {handle}')
+
+        if handle:
```

Comment:
Is there a case where this runs when username != "+remove"?

Reply:
when they enter the same username they were already linekd to

Reply:
which one, the one immediately above checks if the user is valid on dmoj